lib/server/monalisa-synchronization/MonALISAClient.js

```diff
@@ -157,6 +181,104 @@ class MonALISAClient {
     _extractDataPassDetailsFromMonALISAPayload(payload) {
         return { runNumbers: [...new Set(Object.values(payload).map(({ run_no }) => run_no))] };
     }
+
+    /**
+     * Get data passes from MonALISA associated with at least one data pass
+     * @return {{properties: SimulationPass, associations: SimulationPassAssociations}[]} simulation passes list
+     */
+    async getSimulationPasses() {
+        const payload = await this._fetchSimulationPasses();
+        return this._extractSimulationPassesFromPayload(payload);
+    }
+
+    /**
+     * Fetch simulation passes from MonALISA
+     * @return {object<string, Partial<MonALISASimulationPass>>} raw data acquired from MonALISA:
+     * mapping of a simulation pass name to its parameters
+     * @private
+     */
+    async _fetchSimulationPasses() {
+        const { default: fetch } = await import('node-fetch');
+        const specificURL = new URL(this.simulationPassesUrl);
+        specificURL.searchParams.set('res_path', 'json');
+        const response = await fetch(specificURL, this.httpOptions);
+        return response.json();
+    }
+
+    /**
+     * Extract simulation passes from MonALISA response payload
+     * @param {object<string, Partial<MonALISASimulationPass>>} payload raw data acquired from MonALISA:
+     * mapping of a simulation pass name to its parameters
+     * @return {{properties: SimulationPass, associations: SimulationPassAssociations}[]} adapted data
+     * @private
+     */
+    _extractSimulationPassesFromPayload(payload) {
+        return Object.entries(payload).map(([simulationPassName, simulationPassParameters]) => ({
+            ...simulationPassParameters,
+            name: simulationPassName,
+        }))
+            .filter(({ anchor_production, anchor_pass }) => anchor_production.trim() && anchor_pass.trim())
+            .map((simulationPass) => this._monALISASimulationPassToEntity(simulationPass))
+            .filter(({ associations: { lhcPeriods, dataPassesSuffixes } }) => lhcPeriods.length > 0 && dataPassesSuffixes.length > 0);
+    }
+
+    /**
+     * Extract jira id from simulation pass description
+     * @param {string} description simulation pass description
+     * @return {{extractedJiraId: string, extractedDescription: string}} jira id and lighten description
+     */
+    _ectractJiraIdFromDescription(description) {
+        const tokens = description?.trim(0).split(',');
+        if (!tokens) {
+            return null;
+        }
+        const presumedJiraId = tokens[tokens.length - 1]?.trim();
+        if (/[A-Z]+[A-Z0-9]+-\d+/.test(presumedJiraId)) {
```

Comment:
For me, this can be done in a simpler way (I may be wrong):

```js
if (/, [A-Z]+[A-Z0-9]+-\d+/.test(description)){
    const decriptionItems = description.split(',');
    const jiraId = descriptionItems.pop().trim();
    return {
        extractedJiraId: jiraId,
        extractedDescription: descriptionItems.join(',');
    }
}
```

Reply:
applied